pkg/quota_management/account.go

```diff
@@ -1,20 +1,62 @@
 package quota_management
 
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+)
+
 type Account struct {
-	Username            string `yaml:"username"`
-	MaxAllowedInstances int    `yaml:"max_allowed_instances"`
+	Username            string    `yaml:"username"`
+	MaxAllowedInstances int       `yaml:"max_allowed_instances"`
+	GrantedQuota        QuotaList `yaml:"granted_quota,omitempty"`
+}
+
+var _ QuotaManagementListItem = &Account{}
+
+func (account Account) IsInstanceCountWithinLimit(instanceTypeID string, billingModelID string, count int) bool {
+	return count <= account.GetMaxAllowedInstances(instanceTypeID, billingModelID)
+}
+
+func (account Account) GetMaxAllowedInstances(instanceTypeID string, billingModelID string) int {
+	bm, ok := account.getBillingModel(instanceTypeID, billingModelID)
+
+	if !ok {
+		return 0
+	}
+
+	if bm.Allowed <= 0 {
```

Comment:
Looking at this and at the defined configuration format I find it a little bit confusing:
If a user explicitly sets allowed at billing model to 0 I would expect that it means "I don't want to allow quota for this instance type in this billing model, overwriting whatever there is at the upper level".
Another reason is that it is inconsistent with the behavior of max_allowed_instnaces at the global level, where 0 means not allowed.
I think we should consider allowing differentiating between not being specified with being specified to 0.

Reply:
I agree, but this is how it worked until now: you can define max_allowed_instance at `org` level, but if it is `0`, the default value is used. Now we just added one more level (billing model).

Reply:
It is not adding one more level, but different behavior:
At the top level, 0 or unset means "use a default value"
In this new level, 0 or unset means "inherit from the parent level"